pymmcore_plus/core/_mmcore_plus.py

```diff
@@ -157,7 +158,13 @@ def getCameraChannelNames(self) -> Tuple[str, ...]:
             for i in range(self.getNumberOfCameraChannels())
         )
 
-    def run_mda(self, sequence: MDASequence) -> None:
+    def run_mda(self, sequence: MDASequence, writer: MDAWriter = None) -> None:
+
+        if writer is not None:
+            for e in dir(self.events):
+                if "sequence" in e or e == "frameReady":
+                    getattr(self.events, e).connect(getattr(writer, e))
+
```

Comment:
I ended up using the events rather than specifically calling the methods of the writer. This felt cleaner because otherwise you are constantly checking `if writer is not None`. Although I suppose alternatively we could start with:
```python
if writer is None:
    writer = DummyWriter()
````

Reply:
i like this pattern... it's actually pretty general though, so I wonder if perhaps we should make a context manager that lets you temporarily connect an "event listener"?

```py
# or core.events_listener():
with core.events.listener(MDAWriter()):
    core.run_mda(seq)
```